### PR TITLE
Added support for TenableAD license APIs

### PR DESCRIPTION
**Pull request by tushar-balwani**
# Description

Added `Tenable.AD License APIs`

- Get license singleton. (_details_)
- Create new license singleton (_create_)

_updated docs to support license api details_

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added tests for license endpoints to test responses
- [x] Added schema tests for testing payload inputs and response dict validation

**Test Configuration**:
* Python Version(s) Tested: 3.8.6
* Tenable.sc version (if necessary):

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tenable/ad/ldap_configuration/schema.py [hunk 1]
```diff
@@ -25,4 +26,4 @@ class LDAPConfigurationSchema(CamelCaseSchema):
 
     @pre_load
     def convert(self, data, **kwargs):
-        return convert_keys_to_camel(data, special=['search_user_dn'])
+        return convert_keys_to_camel(data, special=[SEARCH_USER_DN])
```
Comment (SteveMcGrath) [inline tenable/ad/ldap_configuration/schema.py:29]: no longer necessary per #526

File: tenable/ad/constants.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+'''
+Constants
+'''
+
+# field-names constants that are required to be converted to
+# last word uppercase
+SEARCH_USER_DN: str = 'search_user_dn'  # ad-ldap-configuration
+EXPIRATION_DATE_UTC: str = 'expiration_date_utc'  # ad-license
```
Comment (SteveMcGrath) [inline tenable/ad/constants.py:8]: no longer necessary per #526

**Review by SteveMcGrath — CHANGES_REQUESTED** [review]

File: tenable/ad/license/api.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            The license object
+
+        Example:
+            >>> tad.license.create(
+            ...     ad_license='license'
+            ... )
+        '''
+        payload = self._schema.dump(self._schema.load({
+            'license': ad_license
+        }))
```
[the remark this answers is not in the document]
Comment (tushar-balwani) [reply tenable/ad/license/api.py]: implemented suggested changes

File: tests/ad/license/test_license_api.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                      'customerName': 'pytenable',
+                      'maxActiveUserCount': 100,
+                      'currentActiveUserCount': 0,
+                      'expirationDateUTC': '2021-11-17T13:44:24.259Z',
+                      'inAppEula': False,
+                      'features': ['something'],
+                      'type': 'license type'
+                  }
+                  )
+    resp = api.license.create(ad_license='license')
```
[the remark this answers is not in the document]
Comment (tushar-balwani) [reply tests/ad/license/test_license_api.py]: implemented suggested changes

File: tenable/ad/ldap_configuration/schema.py (hunk 1 above)
Comment (tushar-balwani) [reply tenable/ad/ldap_configuration/schema.py]: removed pre_load method

File: tenable/ad/constants.py (hunk 2 above)
Comment (tushar-balwani) [reply tenable/ad/constants.py]: removed constants file

**Review by SteveMcGrath — APPROVED** [review]

**Pull request merged**